runtime/sema/type.go

```diff
@@ -7825,6 +7837,12 @@ func checkSubTypeWithoutEquality(subType Type, superType Type) bool {
 					IsSubsetOf(typedSubType.EffectiveInterfaceConformanceSet())
 			}
 
+			// STRINGERTODO: other options? how to make existing simple types
+			// conform to an intersection type?
```

Comment:
It isn't so much about conforming these types to an intersection type, but an interface, `Stringer`.

At the moment we don't have a means to specify a subtype relationship for simple types, so we might want to add that, i.e. generalize this solution a bit.

Note that the intersection type might contain multiple interfaces, e.g. might not just be `{Stringer}`, so checking the first element is not sufficient.

There's also no need, and it is not sufficient, to compare the names only, it is better to check for pointer equality here.

Reply:
> At the moment we don't have a means to specify a subtype relationship for simple types, so we might want to add that, i.e. generalize this solution a bit.

In [5f16333](https://github.com/onflow/cadence/pull/3594/commits/5f163333eb1290df11bd7e1443b08ea03eb8a3c0), I added the functionality for simple types to conform to an interface.

Not sure the best approach in general for all string convertible types to conform to `Stringer` since there are types such as `TheAddressType` and `NumberType` which are not `SimpleType`s so I've left the edge case check in for those.

Reply:
Added a `ConformingType` interface as suggested by https://github.com/onflow/cadence/pull/3594#discussion_r1785276861.